python-checks/src/test/java/org/sonar/python/checks/BooleanCheckNotInvertedCheckTest.java

```diff
@@ -54,30 +57,135 @@ public void operatorStringTest() {
+      "        b = 10\n" +
+      "    return \"item1\" \"item2\"";
+    codeFixed = "" +
+      "def func():\n" +
+      "    if a != 2 and b == 9:\n" +
+      "        b = 10\n" +
+      "    return \"item1\" \"item2\"";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "" +
+      "def func():\n" +
+      "    if a != 2 and not b == 9:\n" +
+      "        b = 10\n" +
+      "    return \"item1\" \"item2\"";
+    codeFixed = "" +
+      "def func():\n" +
+      "    if a != 2 and b != 9:\n" +
+      "        b = 10\n" +
+      "    return \"item1\" \"item2\"";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+  }
+
+  @Test
+  public void not_parentheses() {
+    String codeWithIssue = "a = not ((((((b > c))))))";
+    String codeFixed = "a = b <= c";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not (not (b == c))";
+    codeFixed = "a = not (b != c)";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not (a is (not b))";
+    codeFixed = "a = a is not (not b)";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not (1 == b == c) == 2";
+    codeFixed = "a = (1 == b == c) != 2";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not (1 == b == c) == (d == 2)";
+    codeFixed = "a = (1 == b == c) != (d == 2)";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not ((1 == b == c) and a) == (d == 2)";
+    codeFixed = "a = ((1 == b == c) and a) != (d == 2)";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not (b < foo())";
+    codeFixed = "a = b >= foo()";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+  }
+
+  @Test
+  public void expression_list() {
+    String codeWithIssue = "a = not [] < (c,d)";
+    String codeFixed = "a = [] >= (c, d)";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not [a, (a,b)] is c";
+    codeFixed = "a = [a, (a, b)] is not c";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not (foo((1)) < c)";
+    codeFixed = "a = foo((1)) >= c";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not (foo(()) < c)";
+    codeFixed = "a = foo(()) >= c";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not (() in c)";
+    codeFixed = "a = () not in c";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not ((1,) in c)";
+    codeFixed = "a = (1,) not in c";
+    PythonQuickFixVerifier.verify(check, codeWithIssue, codeFixed);
+
+    codeWithIssue = "a = not ((1,2,3,4) in c)";
+    codeFixed = "a = (1, 2, 3, 4) not in c";
```

Comment:
Yes, we've decided to modify the formatting to follow PEP-8 conventions, if possible and if not too complex for each case.
For this check, it has been decided to rewrite the above lines so that we don't add spacing after the commas